### PR TITLE
Create function for SecretsCreator

### PR DESCRIPTION
**Pull request by iamkirkbater**
Abstracts away the Secrets creation logic into a separate function.

### CODE REVIEW

File: pkg/controller/account/account_controller.go [hunk 1]
```diff
@@ -222,21 +222,11 @@ type secretInput struct {
 	SecretName, NameSpace, awsCredsUserName, awsCredsSecretIDKey, awsCredsSecretAccessKey string
 }
 
-// SRESecretInput is a struct that holds data required to create a new secret CR for SRE admins
-type SRESecretInput struct {
-	SecretName, NameSpace, awsCredsSecretIDKey, awsCredsSecretAccessKey, awsCredsSessionToken, awsCredsConsoleLoginURL string
-}
-
 // SREConsoleInput is a struct that holds data required to create a new secret CR for SRE admins
 type SREConsoleInput struct {
```
Comment (fahlmant) [inline pkg/controller/account/account_controller.go]: This is also input for a secret, any reason not to move it as well?

File: pkg/controller/account/account_controller.go [hunk 2]
```diff
@@ -763,26 +753,6 @@ func CreateAccount(reqLogger logr.Logger, client awsclient.Client, accountName,
...
-		Data: map[string][]byte{
-			"aws_access_key_id":     []byte(input.awsCredsSecretIDKey),
-			"aws_secret_access_key": []byte(input.awsCredsSecretAccessKey),
-			"aws_session_token":     []byte(input.awsCredsSessionToken),
-		},
-	}
-
-}
-
 func (input SREConsoleInput) newConsoleSecret() *corev1.Secret {
```
Comment (fahlmant) [inline pkg/controller/account/account_controller.go]: Should this be moved as well?

File: pkg/controller/account/secrets.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SRESecretInput is a struct that holds data required to create a new secret CR for SRE admins
+type SRESecretInput struct {
+	SecretName, NameSpace, awsCredsSecretIDKey, awsCredsSecretAccessKey, awsCredsSessionToken, awsCredsConsoleLoginURL string
+}
+
+func (input SRESecretInput) newSTSSecret() *corev1.Secret {
```
Comment (fahlmant) [inline pkg/controller/account/secrets.go]: Is this function still used?

Comment (iamkirkbater) [reply pkg/controller/account/secrets.go]: I plan on taking out this function.  I just wanted to make sure it was alright to abstract away the individual implementations of the secret structs and functions into a single one that just takes the map before I finished up removing all of the other secret references.

**Comment by clcollins** [comment]
/lgtm

Also tested locally with byoc/account creation/federated access, all working fine.

**Comment by rogbas** [comment]
@fahlmant @rogbas @iamkirkbater @ArielLima please test locally before we can merge that PR.

**Comment by ArielLima** [comment]
@iamkirkbater can we get a rebase?

**Comment by rogbas** [comment]
@jharrington22 could you please take a look at that PR?

**Comment by iamkirkbater** [comment]
@jharrington22 When I went to rebase, out of the four commits one was a carry over commit that's already been merged, one was a WIP commit, and one was a "fix" commit, so I rebased into a single commit anyway.

I tested after the rebase with all of the `make test-*` commands I saw, and all completed successfully.

**Comment by clcollins** [comment]
Rebased this from master locally, and tested with the `make test-all` target, with no issues.  Watched as secrets were created (though the tests would have caught that too).

/lgtm

**Comment by rogbas** [comment]
@iamkirkbater could you please rebase and push changes again?

**Comment by iamkirkbater** [comment]
This should be all set and ready for final testing and review. @jharrington22 @rogbas @fahlmant

**Pull request merged**